data_processing/scanManager/extractRadiomics.py

```diff
@@ -50,43 +37,70 @@ def cli(cohort_id, container_id, method_id):
 
     if not input_nodes or len (input_nodes)==0:
         logger.error ("Scan is not ready for radiomics (missing annotation?)")
-        logger.info (f"""MATCH (object:scan)-[:HAS_DATA]-(image:mhd) MATCH (object:scan)-[:HAS_DATA]-(label:mha) WHERE id(object)={container_id} RETURN object.SeriesInstanceUID, image.path, label.path""")
-        return 
+        return [] 
+    else:
+        return input_nodes[0].data()
+
+def get_method_data(method_id):
+    with open(f'{method_id}.json') as json_file:
+        method_config = json.load(json_file)['params']
+    return method_config
+
+def add_container_data(container_id, n_meta):
+    conn = Neo4jConnection(uri=os.environ["GRAPH_URI"], user="neo4j", pwd="password")
```

Comment:
Perhaps we can move connection closing outside of query(), and have us manually close() when we are done with the graph connection. This way, we don't need to establish the connection multiple times in the same process.

Reply:
Agreed, managing the DB connection correctly would be a good next step!